pkg/commands/image/list_filter.go

```diff
@@ -0,0 +1,129 @@
+package image
+
+import (
+	corev1alpha1 "github.com/pivotal/kpack/pkg/apis/core/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"regexp"
+	"strings"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+const (
+	filterKindBuilder        = "builder"
+	filterKindClusterBuilder = "clusterbuilder"
+	filterKindLatestReason   = "latest-reason"
+	filterKindStatus         = "ready"
+)
+
+type filter struct {
+	kind   string
+	values []string
+}
+
+func filterImageList(images *v1alpha1.ImageList, flags []string) *v1alpha1.ImageList {
+	filters := parseFilters(flags)
+	if len(filters) == 0 {
+		return images
+	}
+
+	var filteredItems []v1alpha1.Image
+	for _, item := range images.Items {
+		if matchesAll(item, filters) {
+			filteredItems = append(filteredItems, item)
+		}
+	}
+
+	images.Items = filteredItems
+	return images
+}
+
+func parseFilters(flags []string) []filter {
```

Comment:
wondering if it makes sense to be validating filter values? especially, since we call out the supported filters and the supported values for the filters in the command's help text
example of typo scenarios, where we could/should be erroring
```
$ kp image list --filter read=true
NAME           READY    LATEST REASON    LATEST IMAGE                                                                                                                     NAMESPACE
test-image     True     TRIGGER          gcr.io/cf-build-service-dev-219913/ssuresh/test-image@sha256:d939743935ef4bb484de89a3781565b78ccbc4791d72362f7f96c2bf121a9495    default
test-image2    True     CONFIG           gcr.io/cf-build-service-dev-219913/ssuresh/test-image@sha256:d939743935ef4bb484de89a3781565b78ccbc4791d72362f7f96c2bf121a9495    default

$ kp image list --filter latest-reason=stock
Error: no images found
```

Reply:
Normally I'd flag this a feature creep. But validating _the keys_ (scenario 1) resonated with me enough to do a little ditty on it. If you still feel strongly about validating the values (scenario 2), let's make a separate issue about it.